server/src/modules/cats/cats.service.ts

```diff
@@ -11,11 +12,25 @@ export class CatsService {
   ) { }
 
   async create(createCatDto: CreateCatDto): Promise<Cat> {
+
     return this.catsRepository.create(createCatDto);
   }
 
-  async findAll(): Promise<Cat[]> {
-    return this.catsRepository.findAll({ include: [{ model: Mouse, as: 'mice' }] });
+  async findAll(filter: string): Promise<Cat[]> {
+    const whereCondition = filter
+      ? {
+        [Op.or]: [
+          { firstName: { [Op.iLike]: `%${filter}%` } },
+          { lastName: { [Op.iLike]: `%${filter}%` } },
+          { '$mice.name$': { [Op.iLike]: `%${filter}%` } },
+        ],
+      }
+      : {};
+
+    return this.catsRepository.findAll({
+      where: whereCondition,
+      include: [{ model: Mouse, as: 'mice' }]
+    });
```

Comment:
remember i told you about separate and subquery?
given there is more than mice connected to cat this can cause performance issues